workflow-ui/core-android/src/main/java/com/squareup/workflow1/ui/ScreenViewHolder.kt

```diff
@@ -0,0 +1,91 @@
+package com.squareup.workflow1.ui
+
+import android.view.View
+import com.squareup.workflow1.ui.ScreenViewHolder.Companion.Showing
+import com.squareup.workflow1.ui.ScreenViewHolder.Companion.ShowingNothing
+
+/**
+ * Associates a [view] with a function ([runner]) that can update it to display instances
+ * of [ScreenT]. Also holds a reference to the [ViewEnvironment][environment] that was
+ * most recently used to update the [view].
+ *
+ * [environment] should always hold a reference to the [Screen] most recently shown
+ * in [view], with the key [Showing]. [ScreenViewHolder.showing] provides easy access
+ * to it. Note that the shown [Screen] may not be of type [ScreenT], if this
+ * [ScreenViewHolder] is wrapped by another one. (See [ScreenViewFactory.unwrapping].)
+ *
+ * Do not call [runner] directly. Use [ScreenViewHolder.show] instead. Or most commonly,
+ * allow [WorkflowViewStub.show] to call it for you.
+ */
+@WorkflowUiExperimentalApi
+public interface ScreenViewHolder<in ScreenT : Screen> {
```

Comment:
Is the plan to make this type as sealed as possible? I'm asking because the polymorphism is extracted out to the runner. 
If the answer is yes this could be a class, or a sealed class at least, or something. Definitely not important.

But maybe worth mentioning in the docs:
- holder: pretty static / fixed / controlled / not-instantiable from outside, etc
- runner: the "strategy" (strategy pattern) to implement a particular holder.

Reply:
I kept bouncing back and forth btw. using an interface or just making `RealScreenViewHolder` public (and dropping the `Real` prefix, obv.). I forget what finally drove me back to the interface, something related to ease of wrapping. Unclear to me that it's still necessary, I'll take one more look at dropping

Reply:
I _think_ I can pull this off, but I'd prefer to merge what I have and tackle it in a follow up. In the meantime I'll try to punch up the docs as you suggest.

Reply:
The important thing about the interface approach is that it makes it easy to wrap a `ScreenViewHolder` and delegate to the `environment` of the wrapped instance, which is crucial for our crazy `showing` semantics. I suppose we could achieve that another way, but I don't really see what the benefit would be.

This is the case that drove me to write `EnvironmentScreenAndroidIntegrationTest` last week.